prql-compiler/src/parser/mod.rs

```diff
@@ -688,76 +782,69 @@ mod test {
             Literal:
               Integer: 2
-        "###);
-        Ok(())
+        "###)
     }
 
     #[test]
-    fn test_parse_string() -> Result<()> {
-        let double_quoted_ast = expr_of_string(r#"" U S A ""#, Rule::string)?;
+    fn test_parse_string() {
+        let double_quoted_ast = expr_of_string(r#"" U S A ""#).unwrap();
         assert_yaml_snapshot!(double_quoted_ast, @r###"
         ---
         Literal:
           String: " U S A "
         "###);
 
-        let single_quoted_ast = expr_of_string(r#"' U S A '"#, Rule::string)?;
+        let single_quoted_ast = expr_of_string(r#"' U S A '"#).unwrap();
         assert_eq!(single_quoted_ast, double_quoted_ast);
 
         // Single quotes within double quotes should produce a string containing
         // the single quotes (and vice versa).
-        assert_yaml_snapshot!(expr_of_string(r#""' U S A '""#, Rule::string)? , @r###"
+        assert_yaml_snapshot!(expr_of_string(r#""' U S A '""#).unwrap(), @r###"
         ---
         Literal:
           String: "' U S A '"
         "###);
-        assert_yaml_snapshot!(expr_of_string(r#"'" U S A "'"#, Rule::string)? , @r###"
+        assert_yaml_snapshot!(expr_of_string(r#"'" U S A "'"#).unwrap(), @r###"
         ---
         Literal:
           String: "\" U S A \""
         "###);
 
-        assert!(expr_of_string(r#"" U S A"#, Rule::string).is_err());
-        assert!(expr_of_string(r#"" U S A '"#, Rule::string).is_err());
+        expr_of_string(r#"" U S A"#).unwrap_err();
+        expr_of_string(r#"" U S A '"#).unwrap_err();
 
-        // Escapes get passed through (the insta snapshot has them escaped I
-        // think, which isn't that clear, so repeated below).
-        let escaped_string = expr_of_string(r#"" \U S A ""#, Rule::string)?;
+        let escaped_string = expr_of_string(r#"" \nU S A ""#).unwrap();
         assert_yaml_snapshot!(escaped_string, @r###"
         ---
         Literal:
-          String: " \\U S A "
+          String: " \nU S A "
         "###);
-        assert_eq!(
-            escaped_string
-                .kind
-                .as_literal()
-                .unwrap()
-                .as_string()
-                .unwrap(),
-            r#" \U S A "#
-        );
+        let escaped_string = escaped_string
+            .kind
+            .as_literal()
+            .unwrap()
+            .as_string()
+            .unwrap();
+        assert_eq!(escaped_string, " \nU S A ");
 
         // Currently we don't allow escaping closing quotes — because it's not
         // trivial to do in pest, and I'm not sure it's a great idea either — we
         // should arguably encourage multiline-strings. (Though no objection if
         // someone wants to implement it, this test is recording current
         // behavior rather than maintaining a contract).
-        let escaped_quotes = expr_of_string(r#"" Canada \""#, Rule::string)?;
+        let escaped_quotes = expr_of_string(r#"" Canada \"""#).unwrap();
```

Comment:
FWIW I think we can delete this block; it's not needed now we're escaping quotes